go.mod

```diff
@@ -19,37 +19,37 @@ require (
 	github.com/spf13/viper v1.10.1
 	github.com/stretchr/testify v1.8.1
 	github.com/tommy351/goldga v0.5.0
-	go.etcd.io/etcd/api/v3 v3.5.5
-	go.etcd.io/etcd/client/v3 v3.5.5
+	go.etcd.io/etcd/api/v3 v3.5.7
+	go.etcd.io/etcd/client/v3 v3.5.7
 	go.uber.org/atomic v1.11.0
-	google.golang.org/grpc v1.56.3
-	google.golang.org/protobuf v1.30.0
-	k8s.io/api v0.26.4
-	k8s.io/apimachinery v0.26.4
-	k8s.io/client-go v0.26.4
-	knative.dev/pkg v0.0.0-20230502134655-db8a35330281
+	google.golang.org/grpc v1.58.3
+	google.golang.org/protobuf v1.31.0
+	k8s.io/api v0.27.6
+	k8s.io/apimachinery v0.27.6
+	k8s.io/client-go v0.27.6
```

Comment:
There is a similar issue than CI reported saying that the issue is fixed on 0.27.2, however the version here is 0.27.6:
https://github.com/operator-framework/operator-sdk/issues/6396
https://github.com/kubernetes-sigs/controller-runtime/issues/2302

might need to bump to 0.28.x instead.

Reply:
For now pinning with 0.26.4 has fixed the problem.

```
	// pin to 0.26.4 to avoid https://github.com/kubernetes-sigs/controller-runtime/issues/2302
	k8s.io/api => k8s.io/api v0.26.4
	k8s.io/client-go => k8s.io/client-go v0.26.4
	k8s.io/code-generator => k8s.io/code-generator v0.26.4
```

Reply:
Yup, sadly.